src/include/sof/compiler_info.h

```diff
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Karol Trzcinski <karolx.trzcinski@linux.intel.com>
+ */
+
+/**
+ * \file include/sof/compiler_info.h
+ * \brief Compiler version and name descriptor
+ * \author Karol Trzcinski <karolx.trzcinski@linux.intel.com>
+ */
+
+#ifndef __COMPILER_INFO_H__
+#define __COMPILER_INFO_H__
+
+#if defined(__XCC__)
+#define CC_MAJOR (XTHAL_RELEASE_MAJOR / 1000)
+#define CC_MINOR ((XTHAL_RELEASE_MAJOR % 1000) / 10)
+#define CC_MICRO XTHAL_RELEASE_MINOR
+#define CC_NAME "XCC"
+/* __XCC__ */
+#elif defined(__GNUC__) && !defined(__XCC__)
+#define CC_MAJOR __GNUC__
+#define CC_MINOR __GNUC_MINOR__
+#define CC_MICRO __GNUC_PATCHLEVEL__
+#define CC_NAME "GCC"
+/* __GNUC__ */
+#else
+#error "Undefined compiler"
```

Comment:
Not very helpful message. We probably want to tell users that the code is not currently validated on other compilers atm. e.g. "Your compiler is currently unsupported for building SOF. Please validate your compiler and update this file."

Reply:
Ok